src/components/forms/forms.js

```diff
@@ -13,80 +21,122 @@ import {
-  NUMBER: { defaultValue: '', component: formInputGeneric },
+  MULTITEXT: {
+    defaultValue: [],
+    component: formInputMultiText,
+    requiredProps: [],
+    optionalProps: [
+      'placeholder',
+      'ariaDescribedby',
+      'inputStyle'
+    ],
+  },
+  SLIDER: {
+    defaultValue: false,
+    component: formInputSlider,
+    requiredProps: [],
+    optionalProps: [
+      'toggleText',
+    ],
+  },
+  RADIO: {
+    defaultValue: null,
+    component: formInputRadioGroup,
+    requiredProps: [
+      'options',
+      // 'options' example:
+      // [
+      //  {name: 'opt_1', text: 'Option 1'},
+      //  {name: 'other', text: 'Other', type: 'string'} <-- will have textbox
+      // ]
+    ],
+    optionalProps: [
+      'ariaDescribedBy',
+    ],
+    customPropValidation: customRadioPropValidation,
+  },
+  TEXT: {
+    defaultValue: '',
+    component: formInputGeneric,
+    requiredProps: [],
+    optionalProps: [
+      'placeholder',
+      'inputStyle',
+      'ariaDescribedby',
+    ],
+  },
+  NUMBER: {
+    defaultValue: '',
+    component: formInputGeneric,
+    requiredProps: [],
+    optionalProps: [
+      'placeholder',
+      'inputStyle',
+      'ariaDescribedby',
+    ],
+  },
   CHECKBOX: {
     defaultValue: (config) => (config?.valueType === 'string' ? '' : false),
-    component: formInputCheckbox
+    component: formInputCheckbox,
+    requiredProps: [],
+    optionalProps: [
+      'placeholder',
+      'inputStyle',
+    ],
   },
   SELECT: {
     defaultValue: (config) => (config?.isMulti ? [] : ''),
-    updateDefaultValue: ({ selectOptions, defaultValue }) => {
-      const isStringArr = isString(selectOptions[0]);
-      return isStringArr
-        ? { key: defaultValue, displayValue: defaultValue }
-        : defaultValue;
-    },
-    component: formInputSelect
+    updateDefaultValue: updateSelectDefaultValue,
+    component: formInputSelect,
+    requiredProps: [
+      'selectOptions'
+      // 'selectOptions' example:
+      // [
+      //  {displayText: 'Option 1'}
+      //   ^^^ can pass other fields in as extra info, e.g.:
+      //  {displayText: 'Dac 1', dacId: 213}
+      // ]
+    ],
+    optionalProps: [
+      'isCreatable', // allows user to input their own
+      'isMulti',
+      'placeholder',
+      'ariaDescribedby',
+      'required',
+      'selectConfig',
+    ],
+    customPropValidation: customSelectPropValidation,
   },
 };
 
 export const FormValidators = {
-  REQUIRED: {
-    isValid: (value) => value !== undefined && value !== null && value !== '',
-    msg: 'Please enter a value'
-  },
-  URL: {
-    isValid: (val) => {
-      try {
-        new URL(val);
-      } catch (_) {
-        return false;
-      }
-      return true;
-    },
-    msg: 'Please enter a valid url (e.g., https://www.google.com)'
-  },
-  EMAIL: {
-    isValid: isEmailAddress,
-    msg: 'Please enter a valid email address (e.g., person@example.com)'
-  }
+  REQUIRED: requiredValidator,
```

Comment:
Instead of exporting all of these functions from formUtils, can we just put the entire FormValidators object there please?

Reply:
I kind of like them being here, because they demonstrate the functionality of the formfields. To me, a `formUtils` folder contains all of the 'backend' code which a developer who is just trying to use the components doesn't need to know; since it's forward facing, I think it should stay in `Forms.js`. That's just my thought though